src/repr/src/relation.rs

```diff
@@ -208,6 +208,32 @@ impl RelationType {
             (0..self.column_types.len()).collect()
         }
     }
+
+    /// True if the column types match, and keys / nullability are at least as strict.
+    pub fn as_strict_as(&self, other: &RelationType) -> bool {
+        let all_keys = other.keys.iter().all(|key1| {
+            self.keys
+                .iter()
+                .any(|key2| key1.iter().all(|k| key2.contains(k)))
+        });
+        if !all_keys {
+            return false;
+        }
+
+        if self.column_types.len() != other.column_types.len() {
+            return false;
+        }
+
+        for (col1, col2) in self.column_types.iter().zip(other.column_types.iter()) {
+            if col1.nullable && !col2.nullable {
```

Comment:
I read the description of this function and expected this condition to be `if col2.nullable && !col1.nullable`. The description sounds like we want col1 to be at least as strict as col2, i.e. col2's strictness is the predicate. This is just a prose translation, so please disregard if I've misread.

Reply:
I'll try and clean up the prose. The intent is that the any collection described by the `other` type could be described (not incorrectly) by the `self` type. So if you were told the data should be `other`, but it turns out we concluded it is actually `self`, that isn't fundamentally a problem.

Reply:
More text added, and better function name used (well, I think: `subtypes`, which is the correct CS name from the programming languages community).